components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/AbstractHTTPFunction.java

```diff
@@ -90,21 +95,30 @@ protected void executeHttpMethod(HttpUriRequest request, Map<String, Object> eve
                 responseCode = response.getStatusLine().getStatusCode();
                 if (responseCode >= 200 && responseCode < 300) {
                     outcome = Constants.OUTCOME_SUCCESS;
-                    String jsonString = EntityUtils.toString(response.getEntity());
-                    JSONParser parser = new JSONParser();
-                    json = (JSONObject) parser.parse(jsonString);
+                    if (response.getEntity() != null) {
+                        Header contentType = response.getEntity().getContentType();
+                        String jsonString = EntityUtils.toString(response.getEntity());
+                        if (contentType != null && contentType.getValue().contains(TYPE_TEXT_PLAIN)) {
+                            // For 'text/plain', put the response body into the JSON object as a single field.
+                            json = new JSONObject();
+                            json.put("response", jsonString);
```

Comment:
Extract "response" to a constant.

Reply:
Refactored in [053e760](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/140/commits/053e760da0fb8c3528bc8ee40d08eb2b92243320)